Lib/xml/etree/ElementTree.py

```diff
@@ -755,6 +774,10 @@ def write(self, file_or_filename,
             else:
                 encoding = "us-ascii"
         enc_lower = encoding.lower()
+
+        if pretty_print:
+            self._indent(self._root)
```

Comment:
Sorry, but `write()` is an output function. It shouldn't have side-effects that change the tree.
Just imagine you'd print or log the tree for debugging purposes and it would alter the serialised result later on.